### PR TITLE
Add support for a toml config file

### PR DESCRIPTION
**Pull request by rhatdan**
Container Storage is going to be shared by multiple tools.  Therefore
we don't want each one of these tools setting up its own container
storage.  This storage config can be shared between multiple users.
Signed-off-by: Dan Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind Tested with buildah, seems to work well.

File: store.go [hunk 1]
```diff
@@ -1958,11 +1960,64 @@ func stringSliceWithoutValue(slice []string, value string) []string {
...
+		DefaultStoreOptions.GraphDriverName = config.Storage.Driver
+	}
+	if config.Storage.RunRoot != "" {
+		DefaultStoreOptions.RunRoot = config.Storage.RunRoot
+	}
+	if config.Storage.GraphRoot != "" {
+		DefaultStoreOptions.GraphRoot = config.Storage.GraphRoot
+	}
+	for _, s := range config.Storage.Options.AdditionalImageStores {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("%s.imageStore=%s", config.Storage.Driver, s))
```
Comment (nalind) [inline store.go]: This uses different capitalization than what #48 gave to that option.

**Comment by nalind** [comment]
The case-folding behavior of the parser when assigning values to struct members means that we probably don't need any of the toml-specific tagging of structure types, but that's not a blocker.  Except for the capitalization of the `imagestore`/`imageStore` option that it synthesizes, this looks good to me.

**Comment by rhatdan** [comment]
Ok, I changed all of them to be lower case in the config file, and added a man page.

File: storage.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# The "crio" table contains all of the server options.
+[storage]
+
+# Default Storage Driver
+driver = "overlay"
+
+# Temporary storage location
+runroot = "/var/run/containers/storage"
+
+# Primary Read/Write location of containre storage
```
Comment (nalind) [inline storage.conf]: Typo: "containre"

File: storage.conf [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# The "crio" table contains all of the server options.
```
Comment (nalind) [inline storage.conf]: "crio" should be "storage"?

File: store.go [hunk 4]
```diff
@@ -1958,11 +1960,64 @@ func stringSliceWithoutValue(slice []string, value string) []string {
...
+}
+
+// TOML-friendly explicit tables used for conversions.
+type tomlConfig struct {
+	Storage struct {
+		Driver    string                  `toml:"driver"`
+		RunRoot   string                  `toml:"runroot"`
+		GraphRoot string                  `toml:"graphroot"`
+		Options   struct{ OptionsConfig } `toml:"options"`
+	} `toml:"Storage"`
```
Comment (nalind) [inline store.go]: Was this meant to be changed to lower-case as well?

File: storage.conf (hunk 3 above)
Comment (rhatdan) [inline storage.conf]: Fixed

File: storage.conf (hunk 2 above)
Comment (rhatdan) [inline storage.conf]: Fixed

File: store.go [hunk 5]
```diff
@@ -1958,11 +1960,64 @@ func stringSliceWithoutValue(slice []string, value string) []string {
 	return modified
 }
 
+const configFile = "/etc/containers/storage.conf"
+
+// OptionsConfig represents the "crio.api" TOML config table.
```
Comment (nalind) [inline store.go]: "crio" -> "storage.options"?

**Comment by TomSweeneyRedHat** [comment]
LGTM, but I'm not real hip at all with TOML

File: store.go (hunk 5 above)
Comment (rhatdan) [inline store.go]: Fixed,  missed that one.  Lots of cut and paste errors.

**Comment by nalind** [comment]
LGTM, merging.

**Pull request merged**